### ISSUE TITLE
VS2019 Disagrees...

### ISSUE THREAD

**Issue by nurbles** [MicrosoftDocs/cpp-docs#1589]
The final remark, "The traditional Microsoft C++ implementation suppresses a trailing comma if no arguments are passed to the ellipsis." no longer appears to be true as I am getting errors because if the unsuppressed trailing comma when I have no optional parameters to my variadic macros in VS2019.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 17c9f501-4f24-7f53-90f9-c8375802e73f
* Version Independent ID: 0126e007-877f-f21c-266e-4b1fcce0c102
* Content: [Variadic macros](https://docs.microsoft.com/en-us/cpp/preprocessor/variadic-macros?view=vs-2019#feedback)
* Content Source: [docs/preprocessor/variadic-macros.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/preprocessor/variadic-macros.md)
* Product: **visual-cpp**
* Technology: **cpp-language**
* GitHub Login: @mikeblome
* Microsoft Alias: **mblome**

**Comment by mikeblome** [comment]
@nurbles you are presumably seeing that behavior because you have /experimental:preprocessor set. I have updated the topic to mention the new behavior with that option. Should go public in 1-2 days. Thanks for the feedback!

**Comment by nurbles** [comment]
@mikeblome No.  I have never heard of that option.  It is not set in any of my projects, at least, it doesn't show on the "Command line" branch of the C/C++ Configuration properties in my project settings.

For reference, here is a typical set of options from the Command line panel for both debug and release:

Debug:  /JMC /permissive- /MP /FR"x64\Debug\" /GS /analyze- /W4 /Zc:wchar_t /Zi /Gm- /Od /sdl /Fd"x64\Debug\vc142.pdb" /Zc:inline /fp:precise /D "_DEBUG" /D "_WINDOWS" /errorReport:prompt /WX- /Zc:forScope /RTC1 /Gd /MDd /FC /Fa"x64\Debug\" /EHsc /nologo /Fo"x64\Debug\" /Fp"x64\Debug\EV11.pch" /diagnostics:column 

Release:  /permissive- /MP /FR"x64\Release\" /GS /GL /W4 /Gy /Zc:wchar_t /Zi /Gm- /O2 /sdl /Fd"x64\Release\vc142.pdb" /Zc:inline /fp:precise /D "NDEBUG" /D "_WINDOWS" /errorReport:prompt /WX- /Zc:forScope /Gd /Oi /MD /FC /Fa"x64\Release\" /EHsc /nologo /Fo"x64\Release\" /Fp"x64\Release\EV11.pch" /diagnostics:column 

I don't see anything resembling your assumption, sorry.

**Comment by mikeblome** [comment]
@nurbles the product has requested that you create an issue on [C++ Developer Community](https://developercommunity.visualstudio.com/spaces/62/index.html) so that it gets on their radar and they can investigate.

**Comment by nurbles** [comment]
@mikeblome OK, I will submit it as a problem the next time it starts happening.  About an hour after I posted my previous reply I tried to add a new item to a menu resource and got hit by [IDE crash on new menu item](https://developercommunity.visualstudio.com/content/problem/669039/when-try-to-change-the-properties-of-a-menu-item-i.html).  After the crash, pressing F1 while something was selected actually took me to the help for the selected item.
Oh, just to be clear, I had been getting the wrong help for a couple days, which included powering down the computer (**not** hibernating or sleeping!) at the end of the day and starting it and the VS2019 IDE the next day.  So something that happened as part of the crash recovery is what actually corrected the help.
